packages/server/src/leaderboard/calculate.ts

```diff
@@ -122,14 +131,24 @@ export default ({
     }
   }
 
-  const userCompare = (a: InternalUserInfo, b: InternalUserInfo) => {
+  const userCompare = (a: IntermediateUserInfo, b: IntermediateUserInfo) => {
     // sort the users by score
-    // if two user's scores are the same, sort by last solve time
+    // if two user's scores are the same, sort by last tiebreakEligible solve time
+    // if neither user has any tiebreakEligible solves, sort by last solve time
     const scoreCompare = b.score - a.score
     if (scoreCompare !== 0) {
       return scoreCompare
     }
-    return a.lastSolve - b.lastSolve
+    if (
+      a.lastTiebreakEligibleSolve !== undefined ||
+      b.lastTiebreakEligibleSolve !== undefined
+    ) {
+      return (
+        (a.lastTiebreakEligibleSolve ?? Infinity) -
+        (b.lastTiebreakEligibleSolve ?? Infinity)
+      )
+    }
+    return (a.lastSolve ?? Infinity) - (b.lastSolve ?? Infinity)
```

Comment:
This can result in the compare function returning `NaN`, which would make `sort` implementation-defined behavior according to spec. If both have undefined last solve times, we need to make sure to return 0.